packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -119,94 +128,106 @@ describe("GnosisSafe", () => {
       );
       await execTxAction.waitForNonceToUpdate();
 
-      const balanceAfterWithdraw = await provider.getBalance(
-        GNOSIS_SAFE_ADDRESS
-      );
-      expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
+      const balance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+      expect(`${balance}`).to.equal(`${ZERO}`);
     });
+  });
 
-    it("wallet owner should withdraw some ethers", async () => {
+  describe("test cases that needs ERC20 deposit to the wallet", async () => {
+    beforeEach("faucet", async () => {
+      const { address: anaAddress } = anaWallet;
+
+      erc20.setWallet(anaWallet);
+      const mintAction = erc20.mint(GNOSIS_SAFE_ADDRESS, ONE);
+      await mintAction.waitForNonceToUpdate();
+
+      const balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+      expect(`${balance}`).to.equal(`${ONE}`);
+    });
+
+    it("wallet owner should withdraw some ERC20 tokens", async () => {
       const signers = [anaWallet];
+      const tokenAddress = ERC20_ADDRESS;
       const { address: toAddress } = anaWallet;
       const value = ONE;
 
       gnosisSafe.setWallet(anaWallet);
-      const action = await gnosisSafe.transferEther(signers, toAddress, value);
-
-      const onConfirmation = async message => {
-        const { data } = message;
-        const { args } = data;
-
-        action.unsubscribe();
-
-        const balance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
-        expect(`${balance}`).to.equal(`${ZERO}`);
-      };
-
-      const onError = message => {
-        const { error } = message;
-        console.log(error);
-      };
+      const action = await gnosisSafe.transferERC20(
+        signers,
+        tokenAddress,
+        toAddress,
+        value
+      );
+      await action.waitForNonceToUpdate();
```

Comment:
Let's rename this function to `waitForOneConfirmation` soon

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/237

Reply:
Thanks. Resolving